src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/RedundantImportCheck.java

```diff
@@ -156,4 +189,39 @@ private static boolean isFromPackage(String importName, String pkg) {
         return front.equals(pkg);
     }
 
+    /**
+     * Determines if an import statement is a star import.
+     * @param imp the import
+     * @return whether imp is a start import
+     */
+    private static boolean isStarImport(FullIdent imp) {
+        return imp.getText().endsWith(".*");
+    }
+
+    /**
+     * Determines if a package imported by {@code imp} can be imported using {@code starImp}.
+     * @param starImp the star import
+     * @param imp the import
+     * @return whether package imported by {@code imp} can be imported using {@code starImp}
+     */
+    private static boolean isDuplicatedByStarImport(FullIdent starImp, FullIdent imp) {
+        final String importSplitRegexp = "\\.";
+
+        final String[] impSplitted = imp.getText().split(importSplitRegexp);
+        final String[] starImpSplited = starImp.getText().split(importSplitRegexp);
+
+        boolean isEquals = true;
+
+        if (impSplitted.length == starImpSplited.length) {
+            // check for common prefix
+            for (int i = 0; i < impSplitted.length - 1 && isEquals; i++) {
```

Comment:
Is there no way to do the same thing without the split and just do `lastIndexOf` and check for string equality that way.
Wouldn't that be faster especially then the regular expression split?

Reply:
Good point! Fixed